stackslib/src/chainstate/stacks/mod.rs

```diff
@@ -621,13 +619,74 @@ impl_array_hexstring_fmt!(TokenTransferMemo);
 impl_byte_array_newtype!(TokenTransferMemo, u8, 34);
 impl_byte_array_serde!(TokenTransferMemo);
 
+/// Cause of change in mining tenure
+/// Depending on cause, tenure can be ended or extended
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum TenureChangeCause {
+    /// A valid winning block-commit, end current tenure
+    BlockFound = 0,
+    /// No winning block-commits, extend current tenure
+    NoBlockFound = 1,
+    /// A “null miner” won the block-commit (see the MEV solution below)
+    NullMiner = 2,
+}
+
+impl TryFrom<u8> for TenureChangeCause {
+    type Error = ();
+
+    fn try_from(num: u8) -> Result<Self, Self::Error> {
+        match num {
+            0 => Ok(Self::BlockFound),
+            1 => Ok(Self::NoBlockFound),
+            2 => Ok(Self::NullMiner),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct SchnorrThresholdSignature {
+    //pub point: wsts::Point,
+    //pub scalar: wsts::Scalar,
+}
+
+/// Reasons why a `TenureChange` transaction can be de
+pub enum TenureChangeError {
+    SignatureInvalid,
+    /// Not signed by required threshold (>70%)
```

Comment:
This is indistinguishable from `SignatureInvalid`.  If less than 70% of signers sign, then they cannot produce a valid signature.